src/redux/sagas/assistantSaga.jsx

```diff
@@ -599,19 +650,110 @@ const filterSourceCredibilityResults = (originalResult) => {
+      addToRelevantSourceCred(sourceCredibilityDict[link].positive, result);
     } else if (
       result["source-type"] === "mixed" &&
       result["source"] !== "GDI-MMR"
     ) {
-      addToRelevantSourceCred(mixedResult, result);
+      addToRelevantSourceCred(sourceCredibilityDict[link].mixed, result);
     } else if (result["source-type"] === "caution") {
-      addToRelevantSourceCred(cautionResult, result);
+      addToRelevantSourceCred(sourceCredibilityDict[link].caution, result);
     }
   });
-  return [positiveResult, cautionResult, mixedResult];
+
+  // catching the missing links without source credibility results
+  for (let uniqueLink of uniqueLinkList) {
+    if (!sourceCredibilityDict[uniqueLink]) {
+      sourceCredibilityDict[uniqueLink] = {
+        link: uniqueLink,
+        url_color: trafficLightColors.unlabelled,
+        positive: [],
+        mixed: [],
+        caution: [],
+      };
+    }
+  }
+
+  const positiveResults = sourceCredibilityDict[inputUrl].positive.length
+    ? sourceCredibilityDict[inputUrl].positive
+    : null;
+  const mixedResults = sourceCredibilityDict[inputUrl].mixed.length
+    ? sourceCredibilityDict[inputUrl].mixed
+    : null;
+  const cautionResults = sourceCredibilityDict[inputUrl].caution.length
+    ? sourceCredibilityDict[inputUrl].caution
+    : null;
+  delete sourceCredibilityDict[inputUrl];
+
+  return [positiveResults, mixedResults, cautionResults, sourceCredibilityDict];
+};
+
+const sortSourceCredibilityLinks = (
+  sourceCredibilityDict,
+  trafficLightColors,
+) => {
+  if (!sourceCredibilityDict) {
+    return null;
+  }
+
+  let positiveLinks = [];
+  let mixedLinks = [];
+  let cautionLinks = [];
+  let unlabelledLinks = [];
+
+  for (let link in sourceCredibilityDict) {
+    let result = sourceCredibilityDict[link];
+
+    result.positive = result.positive.length ? result.positive : null;
+    result.mixed = result.mixed.length ? result.mixed : null;
+    result.caution = result.caution.length ? result.caution : null;
+
+    if (result.caution) {
+      result.url_color = trafficLightColors.caution;
+      cautionLinks.push(link);
+    } else if (result.mixed) {
+      result.url_color = trafficLightColors.mixed;
+      mixedLinks.push(link);
+    } else if (result.positive) {
+      result.url_color = trafficLightColors.positive;
+      positiveLinks.push(link);
+    } else {
+      result.url_color = trafficLightColors.unlabelled;
+      unlabelledLinks.push(link);
+    }
+  }
+
+  let extractedLinks = [];
+  extractedLinks = cautionLinks.length
+    ? extractedLinks.concat(cautionLinks.sort())
+    : extractedLinks;
+  extractedLinks = mixedLinks.length
+    ? extractedLinks.concat(mixedLinks.sort())
+    : extractedLinks;
+  extractedLinks = positiveLinks.length
+    ? extractedLinks.concat(positiveLinks.sort())
+    : extractedLinks;
+  extractedLinks = unlabelledLinks.length
+    ? extractedLinks.concat(unlabelledLinks.sort())
+    : extractedLinks;
```

Comment:
This should also work as all  you are guaranteed to get an array type for all the individual links category?:

```
extractedLinks = extractedLinks.concat(cautionLinks, mixedLinks, positiveLinks, unlabelledLinks); 
```

Reply:
Yes - done!